hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerProtocolServerSideTranslatorPB.java

```diff
@@ -183,6 +183,9 @@ private OMResponse processRequest(OMRequest request) throws
           OzoneManagerRatisUtils.checkLeaderStatus(ozoneManager);
         }
         try {
+          if (request.hasS3Authentication()) {
+            ozoneManager.setS3Auth(request.getS3Authentication());
+          }
```

Comment:
I think the same is missing for write case in `submitRequestDirectlyToOM`.

Instead of repeating it in 4 places (read/write * ratis/non-ratis), wouldn't it be simpler to wrap the main flow of `processRequest` in a single `try-finally` to set/clear auth?

Reply:
Yes, makes sense. Will make the change.